packages/content-conformance/src/rules/ensure-valid-nav-paths.js

```diff
@@ -0,0 +1,71 @@
+import flat from 'flat'
+
+// Recursively traverse json nav data, and validate node paths against flat list of files from context
+function visitor(navData, filePaths, report) {
+  // flatten nav data file
+  const flatNodes = flat(navData)
+  // get all the paths from the nav data file
+  const navDataPaths = Object.entries(flatNodes)
+    .filter(([key]) => {
+      return key.endsWith('.path')
+    })
+    .map(([, value]) => value)
+
+  navDataPaths.forEach((p) => {
+    const namedFile = `${p}.mdx`
+    const indexFile = `${p}/index.mdx`
+
+    const namedFileExists = filePaths.includes(namedFile)
+    const indexFileExists = filePaths.includes(indexFile)
+
+    if (!namedFileExists && !indexFileExists) {
+      report(
+        `NavData path (${p}) is missing a corresponding file in the content directory`
+      )
+    }
+
+    if (namedFileExists && indexFileExists) {
+      report(
+        `NavData path (${p}) should not have both a named file and an index file in the content directory`
+      )
+    }
+  })
+}
+
+/** @type {import('../types.js').ConformanceRuleBase} */
+export default {
+  id: 'ensure-valid-nav-paths',
+  type: 'data',
+  description: 'TODO.',
+  executor: {
+    async dataFile(file, context) {
+      // by convention, nav data files are those that end in nav-data.json
+      if (!file.basename.endsWith('nav-data.json')) {
+        return
+      }
+
+      const subpath = file.basename.split('-nav-data.json')[0]
```

Comment:
How will we handle `terraform-plugin-framework` as well as `api-docs`? 💭

Reply:
> `terraform-plugin-framework`

Oh, good catch... we might need maintain an internal mapping :| 

> `api-docs`

Looking at [Vault](https://github.com/hashicorp/vault/tree/main/website/data), the `subpath` looks like it will map correctly to [website/content/api-docs](https://github.com/hashicorp/vault/tree/main/website/content/api-docs)